public/src/components/switchboard.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import get from 'lodash/get';
+import set from 'lodash/set';
+
+import FormLabel from '@material-ui/core/FormLabel';
+import FormControl from '@material-ui/core/FormControl';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+
+
+interface Switches {
+  [key: string]: boolean
+  stripeOneOff: boolean,
+  payPalOneOff: boolean,
+  stripeRecurring: boolean,
+  payPalRecurring: boolean,
+  directDebitRecurring: boolean
+}
+
+function safelyExtractSwitchValue(json: object, path: string): boolean {
+  const switchValue = get(json, path, false);
+  return switchValue === 'On';
+}
+
+function jsonToSwitches(json: object): Switches {
+  return {
+    stripeOneOff: safelyExtractSwitchValue(json, 'value.switches.oneOffPaymentMethods.stripe'),
+    payPalOneOff: safelyExtractSwitchValue(json, 'value.switches.oneOffPaymentMethods.payPal'),
+    stripeRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.stripe'),
+    payPalRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.payPal'),
+    directDebitRecurring: safelyExtractSwitchValue(json, 'value.switches.recurringPaymentMethods.directDebit')
+  };
+}
+
+export class Switchboard extends React.Component {
+  state: Switches;
```

Comment:
nice, does this mean it'll fail to compile if you assign another type to `this.state`?

Reply:
yep. it also enforces that you set a value for it in the constructor